pyntc/devices/asa_device.py

```diff
@@ -155,6 +156,45 @@ def _wait_for_device_reboot(self, timeout=3600):
         # TODO: Get proper hostname parameter
         raise RebootTimeoutError(hostname=self.host, wait_time=timeout)
 
+    def _wait_for_peer_reboot(self, acceptable_states: Iterable[str], timeout: int = 3600) -> None:
+        """
+        Wait for peer device to come online and reach an acceptable state.
+
+        Args:
+            acceptable_states (iter): A list of states that are acceptable for considering peer to be ready.
+            timeout (int): The maximum amount of time to wait for peer device to be online and reach a state in ``acceptable_states``.
+
+        Raises:
+            RebootTimeoutError: When the ``timeout`` value has been reached and the peer has not reached a state in ``acceptable_states``.
+
+        Example:
+            >>> dev = ASADevice(**connection_args)
+            >>> dev.peer_redundancy_state
+            'standby ready'
+            >>> dev.show("failover reload-standby")
+            >>> dev._wait_for_peer_reboot(acceptable_states=["standy ready"])
+            RebootTimeoutError...
+            >>> dev.peer_redundancy_state
+            'cold standby'
+            >>> dev.show("failover reload-standby")
+            >>> dev._wait_for_peer_reboot(acceptable_states=["cold-standby", "standy ready"])
+            >>> dev.peer_redundancy_state
+            'cold standby'
+            >>>
+        """
+        start = time.time()
+        while time.time() - start < timeout:
+            if self.peer_redundancy_state == "failed":
+                break
+
+        while time.time() - start < timeout:
+            if self.peer_redundancy_state in acceptable_states:
+                return
+            time.sleep(1)
```

Comment:
Can this be put into a single while loop? If I read this right, it will start the check, only leave the while loop if it has a state of break. Then it will continue on, but since the time has elapsed it will only enter the RebootTimeOut Error. Never really reaching the second loop. 

I forget if the failed state will appear right away or not. Is that the purpose of the first while loop?

Reply:
The point of this is that the peer device will be dead before the connected device realizes it. Without the first while loop, the second loop will hit the return before the connected device realizes the peer is dead, which gives a false positive. The first while loop waits for the connected device to acknowledge failure of the peer device, then the second loop waits for the peer to come back online.